GoogleForCreators/web-stories-wp
# Apply Page Layouts

## Summary


Render page layouts in pane and select and allow clicking them to to overwrite the currently active page with the contents of the page layout. If the page is not blank, a confirmation is required before applying the page layout.

Includes storybook for 

## Relevant Technical Choices


* Preview pages have been refactored into `edit-story` so that they can be used by both the dashboard and page layouts
* For now, page layouts are not draggable, pending moving to Movable for dragging as seen in #5719

## To-do


* [x] click to apply page layout
* [x] filter out template pages that shouldn't be used as page layouts (deferred to 5687-5690)
* [x] add names to each page layout page (deferred to 5687-5690)
* [x] apply grid pattern in place of images when page layouts are applied (will be follow-up)
* [x] get final copy for the page confirmation dialog if the page already has edits
* [x] storybook for confirmation dialog
* [x] test for duplicate page reducer
* [ ] karma tests for pane

## User-facing changes



## Testing Instructions


* Click the page layouts pane and choose a layout to apply. Animations should run on hover of each page layout.
* Apply a page layout by clicking on the layout in the layouts pane. It should replace all elements on the page and copy all animations from the page layout.
  * If the current page has any modifications from being a blank white page, a confirmation dialog should show up asking whether to clear the page and replace it with the content of the page layout
  * If the page has no edits (blank white), the layout should apply without a confirmation dialog

---



Fixes #3405
